tests/unit/backend/chalice/api_server/test_v1_genesets.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from backend.corpora.common.corpora_orm import CollectionVisibility
+from backend.corpora.common.entities.geneset import GenesetDatasetLink
+from tests.unit.backend.chalice.api_server.base_api_test import BaseAuthAPITest
+from tests.unit.backend.chalice.api_server.mock_auth import get_auth_token
+from tests.unit.backend.fixtures.mock_aws_test_case import CorporaTestCaseUsingMockAWS
+
+
+class TestGenesets(BaseAuthAPITest, CorporaTestCaseUsingMockAWS):
+    def setUp(self):
+        CorporaTestCaseUsingMockAWS.setUp(self)
+
+    def tearDown(self):
+        CorporaTestCaseUsingMockAWS.tearDown(self)
+
+    def _get_geneset_ids(self, collection_id, headers, public=False):
+        if public:
+            rsp = self.app.get(f"/dp/v1/collections/{collection_id}", headers=headers)
+        else:
+            rsp = self.app.get(f"/dp/v1/collections/{collection_id}?visibility=PRIVATE", headers=headers)
+        rsp.raise_for_status()
+        bdy = json.loads(rsp.body)
+        return [g["id"] for g in bdy.get("genesets", [])]
+
+    def test__delete_geneset_ACCEPTED(self):
+        """
+        Delete a geneset from a private collection.
+        """
+
+        headers = dict(host="localhost", Cookie=get_auth_token(self.app))
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE, owner="test_user_id"
+        )
+
+        with self.subTest("With collection"):
```

Comment:
"With Collection" and "With Dataset" seem to share lots of the same test code. To keep things DRY, is it possible to extract the common test code into a function and call them in both "With Collection" and "With Dataset" blocks?

Thanks!

Reply:
keeping it DRY